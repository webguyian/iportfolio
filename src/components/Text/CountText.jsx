import React from 'react';
import PropTypes from 'prop-types';

import Text from 'components/Text/Text';

const CountText = props => {
  const { children, count, emptyLabel, ...otherProps } = props;

  if (count === 1) {
    const label = children.slice(0, children.length - 1);

    return <Text {...otherProps}>{`${count} ${label}`}</Text>;
  }

  return <Text {...otherProps}>{`${count || emptyLabel} ${children}`}</Text>;
};

CountText.propTypes = {
  children: PropTypes.node.isRequired,
  count: PropTypes.number,
  emptyLabel: PropTypes.string
};

CountText.defaultProps = {
  count: 0,
  emptyLabel: 'No'
};

export default CountText;
