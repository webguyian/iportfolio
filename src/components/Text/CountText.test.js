import React from 'react';
import { create } from 'react-test-renderer';

import CountText from './CountText';

describe('<CountText />', () => {
  it('renders correctly', () => {
    const component = create(<CountText>Reminders</CountText>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with one item', () => {
    const component = create(<CountText count={1}>Reminders</CountText>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with multiple items', () => {
    const component = create(<CountText count={12}>Reminders</CountText>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom emptyLabel', () => {
    const component = create(
      <CountText emptyLabel="Zero">Reminders</CountText>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
