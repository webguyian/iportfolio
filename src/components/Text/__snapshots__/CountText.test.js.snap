// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountText /> renders correctly 1`] = `
<span
  className="ui-text ui-text--body"
>
  No Reminders
</span>
`;

exports[`<CountText /> renders correctly with custom emptyLabel 1`] = `
<span
  className="ui-text ui-text--body"
>
  Zero Reminders
</span>
`;

exports[`<CountText /> renders correctly with multiple items 1`] = `
<span
  className="ui-text ui-text--body"
>
  12 Reminders
</span>
`;

exports[`<CountText /> renders correctly with one item 1`] = `
<span
  className="ui-text ui-text--body"
>
  1 Reminder
</span>
`;
